.github/workflows/main.yml

```
@@ -0,0 +1,64 @@
+      - name: Run Tests for All Modules in Parallel
+        run: |
+          find backend/Api/Module -type d -name "*.Test" | xargs -n 1 -P 4 -I {} sh -c 'echo "Running tests for module: {}"; dotnet test "{}"'
+
+  publish:
+    name: Publish Application
+    runs-on: ubuntu-latest
+    needs: build
+
+    steps:
+      - name: Checkout Code
+        uses: actions/checkout@v3
```

actionlint (1.7.4)

the runner of "actions/checkout@v3" action is too old to run on GitHub Actions. update the action's version to fix this issue

(action)